server/src/test/java/io/spine/server/storage/EntityRecordStorageTest.java

```diff
@@ -0,0 +1,603 @@
+            EntityRecord entityRecord = optional.get();
+
+            Message unpacked = unpack(entityRecord.getState());
+            assertFalse(isDefault(unpacked));
+        }
+
+        @SuppressWarnings("MethodWithMultipleLoops")
+        @Test
+        @DisplayName("multiple records according to the given field mask")
+        void multipleRecords() {
+            EntityRecordStorage<StgProjectId, StgProject> storage = storage();
+            int count = 10;
+            List<StgProjectId> ids = new ArrayList<>();
+            @SuppressWarnings("rawtypes")
+            Class<? extends EntityState> stateClass = null;
+
+            for (int i = 0; i < count; i++) {
+                StgProjectId id = newId();
+                EntityState<StgProjectId> state = newState(id);
+                if (stateClass == null) {
+                    stateClass = state.getClass();
+                }
+                EntityRecord record = EntityRecordStorageTestEnv.newStorageRecord(id, state);
+                storage.write(id, record);
+                ids.add(id);
+            }
+
+            int bulkCount = count / 2;
+            FieldMask fieldMask = fromFieldNumbers(stateClass, 2);
+            Iterator<EntityRecord> readRecords = storage.readAll(ids.subList(0, bulkCount),
+                                                                 fieldMask);
+            List<EntityRecord> readList = newArrayList(readRecords);
+            assertThat(readList).hasSize(bulkCount);
+            for (EntityRecord record : readList) {
+                Message state = unpack(record.getState());
+                assertMatchesMask(state, fieldMask);
+            }
+        }
+
+        @Test
+        @DisplayName("archived records if specified")
+        void archivedRecords() {
+            StgProjectId activeRecordId = newId();
+            StgProjectId archivedRecordId = newId();
+
+            EntityRecord activeRecord =
+                    buildStorageRecord(activeRecordId, newState(activeRecordId));
+            EntityRecord archivedRecord =
+                    buildStorageRecord(archivedRecordId, newState(archivedRecordId));
+            TransactionalEntity<StgProjectId, ?, ?> activeEntity = newEntity(activeRecordId);
+            TransactionalEntity<StgProjectId, ?, ?> archivedEntity = newEntity(archivedRecordId);
+
+            archive(archivedEntity);
+
+            EntityRecordStorage<StgProjectId, StgProject> storage = storage();
+
+            storage.write(create(activeEntity, activeRecord));
+            storage.write(create(archivedEntity, archivedRecord));
+
+            StgProject.Query query =
+                    StgProject.query()
+                                               .where(archived.lifecycle(), true)
+                                               .build();
+            assertQueryHasSingleResult(query, archivedRecord, storage);
+        }
+    }
+
+    @Nested
+    @DisplayName("filter records")
+    class Filtering {
```

Comment:
Please consider splitting this long test suite into several ones that would focus on reading, filtering, etc.